packages/protocol/test/signal/SignalService.integration.test.ts

```diff
@@ -0,0 +1,99 @@
+import { expect } from "chai";
+import { ethers } from "hardhat";
+import { TestHeaderSync } from "../../typechain";
+import { deploySignalService, getSignalProof } from "../utils/signal";
+import deployAddressManager from "../utils/addressManager";
+import { getBlockHeader } from "../utils/rpc";
+import { loadFixture } from "@nomicfoundation/hardhat-network-helpers";
+
+describe("integration:SignalService", function () {
+    async function deployIntegrationSignalServiceFixture() {
+        const [owner, nonOwner] = await ethers.getSigners();
+
+        const { chainId } = await ethers.provider.getNetwork();
+
+        const srcChainId = chainId;
+
+        // seondary node to deploy L2 on
+        const l2Provider = new ethers.providers.JsonRpcProvider(
+            "http://localhost:28545"
+        );
+
+        const l1Signer = await ethers.provider.getSigner();
+
+        const l2Signer = await l2Provider.getSigner(
+            (
+                await l2Provider.listAccounts()
```

Comment:
We have a method to get signers already, we should reuse those. In utils folder. Same with providers